lib/ReactViews/Tools/DeltaTool/DeltaTool.jsx

```diff
@@ -0,0 +1,221 @@
+      **Secondary image**: ${secondDateStr}
+    `);
+
+    // item.loadingMessage = "Loading difference map";
+    // item.isLoading = true;
+
+    const firstDateParam = dateParamFormat(primaryDate);
+    const secondDateParam = dateParamFormat(secondaryDate);
+    item.parameters["time"] = `${firstDateParam},${secondDateParam}`;
+
+    item.showDeltaImagery(firstDateParam, secondDateParam);
+    onCloseTool();
+  }
+
+  // Set primary & secondary dates to the last 2 available dates
+  function setDatesFromAvailableDates(availableDates) {
+    const [primaryDate, secondaryDate] = availableDates.slice(
+      availableDates.length - 2
+    );
+    if (primaryDate) {
+      setPrimaryDate(primaryDate);
+      setSecondaryDate(secondaryDate || primaryDate);
+    }
+  }
+
+  return (
+    <div className={Styles.deltaTool}>
+      <h1 className={Styles.title}>Change Detection: {catalogItem.name}</h1>
+      <div className={Styles.body}>
+        <div>
+          <span>
+            This tool visualizes the difference between imagery captured at two
+            discrete points in time.
+          </span>
+          {location && <PrettyLocation location={location} />}
+          {location === undefined ? (
+            <h3>
+              To view available imagery, please select your location of interest
+              on the map opposite.
+            </h3>
+          ) : (
+            <DatePickers
+              item={item}
+              primaryDate={primaryDate}
+              setPrimaryDate={setPrimaryDate}
+              secondaryDate={secondaryDate}
+              setSecondaryDate={setSecondaryDate}
+            />
+          )}
+        </div>
+        <div className={Styles.buttons}>
+          <button className={Styles.cancelBtn} onClick={cancelDeltaTool}>
+            Cancel
+          </button>
+          <button
+            className={Styles.generateDeltaBtn}
+            onClick={generateDelta}
+            disabled={location === undefined}
+          >
+            Generate Difference Map
+          </button>
+        </div>
+      </div>
+      <LocationPicker
+        terria={terria}
+        location={location}
+        onPick={onUserPickLocation}
+        message={pickerMessage}
+      />
+    </div>
+  );
+}
+
+DeltaTool.propTypes = {
+  terria: PropTypes.object.isRequired,
+  tool: PropTypes.object.isRequired,
+  onCloseTool: PropTypes.func.isRequired
+};
+
+DeltaTool.displayName = "DeltaTool";
+
+function PrettyLocation({ location }) {
+  const prettyLocation = prettifyCoordinates(
+    location.longitude,
+    location.latitude
+  );
+  return (
+    <section>
+      <h4>Selected Location</h4>
+      <div className={Styles.location}>
+        {prettyLocation.latitude}, {prettyLocation.longitude}
+      </div>
+    </section>
+  );
+}
+
+PrettyLocation.propTypes = {
+  location: PropTypes.object.isRequired
+};
+
+PrettyLocation.displayName = "PrettyLocation";
+
+function duplicateItem(item) {
```

Comment:
I know it's for ease but we've got the exact same code so might as well avoid duplication here - let's pull out the `splitItem()` fn from ViewingControls, add some controls via parameters/options on whether to do other side effects (`addUserCatalogMember` etc etc)

Reply:
Have extracted the common bits to a new shared import. Not very configurable atm, since much of the things like `addUserCatalogMember` is call site specific.